### PR TITLE
Binomial Dist Edge Parameters & Testing

### PR DESCRIPTION
**Pull request by knappador**
This is a rough draft to trigger discussion.  

Some automated tools and test sanity can easily wind up with p = 1.0 etc or zero-length trial sequences.  The Binomial distribution, while somewhat or totally degenerate in these cases, has clearly correct answers and behavior.  Upstream tools that anticipate these behaviors correctly should be allowed to create edge & corner case distributions.

I only tested the mass function to see if it would cause any zero-division.  The saddle point math could easily have some holes lurking.

### CODE REVIEW

**Comment by knappador** [comment]
I see that some automatic rustfmt decisions filtered into the commit.  Is there yet a policy around use / disuse of rustfmt for source control?  I'll happily edit the commit and either include or exclude the whitespace changes.

File: src/distribution/binomial.rs [hunk 1]
```diff
@@ -494,6 +491,18 @@ mod tests {
             &p,
             1e-14,
         );
+
+        let d2 = new!(1, 1.0);
```
Comment (IvanUkhov) [inline src/distribution/binomial.rs]: Please use just `d` everywhere.

File: src/distribution/binomial.rs [hunk 2]
```diff
@@ -319,26 +319,23 @@ fn approximate_by_normal(p: f64, np: f64, v: f64, u: f64) -> f64 {
...
-    ) +
-    sd_em4 * (
-        w6 * (2.0 * p - 1.0) * (p2 - p + 1.0) * (p2 - p + 19.0) / 34020.0 +
-        w4 * (2.0 * p - 1.0) * (9.0 * p4 - 18.0 * p3 - 35.0 * p2 + 44.0 * p - 25.0) / 15120.0 +
-        w2 * (2.0 * p - 1.0) * (
-                923.0 * p4 - 1846.0 * p3 + 5271.0 * p2 - 4348.0 * p + 5189.0
-        ) / 408240.0 -
-        4.0 * (2.0 * p - 1.0) * (p + 1.0) * (p - 2.0) * (23.0 * p2 - 23.0 * p + 2.0) / 25515.0
-    )
+    np + sd * w + (p + 1.0) / 3.0 - (2.0 * p - 1.0) * w2 / 6.0
```
Comment (IvanUkhov) [inline src/distribution/binomial.rs]: Answering your question, Rustfmt is mandatory. Well, except for a few cases 😀 It’s just that some blocks of code become extremely illegible once formatted, and so I usually cherry-pick rustfmt’s changes. In this case, I think it’s all right. However, I wouldn’t mix formatting changes with conceptual changes. If you don’t mind, at least create a separate commit, and I won’t squash.

**Comment by IvanUkhov** [comment]
I close this as stale.

**Pull request closed without merging**